<launch>
  <arg name="image_topic_name_right" default="image_raw"/>
  <arg name="image_topic_name_left" default="image_raw"/>
  <arg name="lowThreshold" default="50"/>


  <param name="detection_lowerThreshold" value="$(arg lowThreshold)"/>



  <group ns="/stereo/right/">
    <param name="image_topic_name" value="$(arg image_topic_name_right)"/>
    <node name="edge_detector_right" pkg="edge_detection" type="canny_detector" output="screen">
      <remap from="/edgeDetector/edge_map" to="edge_map"/>
      <remap from="/edgeDetector/image_edge_mask" to="image_edge_mask"/>
    </node>
  </group>



  <group ns="/stereo/left/">
    <param name="image_topic_name" value="$(arg image_topic_name_left)"/>
    <node name="edge_detector_left" pkg="edge_detection" type="canny_detector" output="screen">
      <remap from="/edgeDetector/edge_map" to="edge_map"/>
      <remap from="/edgeDetector/image_edge_mask" to="image_edge_mask"/>
    </node>
  </group>

</launch>
